src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -437,6 +575,23 @@ public Activity SetEndTime(DateTime endTimeUtc)
             return this;
         }
 
+        /// <summary>
+        /// Get the context of the activity. Context becomes valid only if the activity has been started.
+        /// otherwise will default context.
+        /// </summary>
+        public ActivityContext Context
+        {
+            get
+            {
+                if (_id == null && _spanId == null)
```

Comment:
The rules about which private fields are populated at which times are a bit messy in the Activity type (lazy initialization for performance). For example the code below assumes that _traceId is always set for W3C ids but there are scenarios where _traceId == null && TraceId != null. I'd suggest avoiding all the lazy field initialization details by implementing this solely in terms of properties:
```suggestion
                return new ActivityContext(TraceId, SpanId, ActivityTraceFlags, TraceStateString);
```

Reply:
good idea!